### PR TITLE
#204 fix: BFS implementation

### PR DESCRIPTION
**Pull request by trungnotchung**
Refactors the `areCausallyRelatedUsingBFS` method to simplify the BFS implementation for checking causal relationships between vertices.

### CODE REVIEW

File: packages/object/src/hashgraph/index.ts [hunk 1]
```diff
@@ -223,47 +223,46 @@ export class HashGraph {
...
 
+	// Time complexity: O(V), Space complexity: O(V)
+	areCausallyRelatedUsingBFS(hash1: Hash, hash2: Hash): boolean {
+		return (
+			this._areCausallyRelatedUsingBFS(hash1, hash2) ||
+			this._areCausallyRelatedUsingBFS(hash2, hash1)
+		);
+	}
+
 	// Time complexity: O(1), Space complexity: O(1)
```
Comment (d-roak) [inline packages/object/src/hashgraph/index.ts:266]:
```suggestion
	areCausallyRelated(hash1: Hash, hash2: Hash): boolean {
		return (
			this._areCausallyRelated(hash1, hash2) ||
			this._areCausallyRelated(hash2, hash1)
		);
	}

```

File: packages/object/src/hashgraph/index.ts [hunk 2]
```diff
@@ -223,47 +223,46 @@ export class HashGraph {
 		return test1 || test2;
 	}
 
-	// Time complexity: O(V), Space complexity: O(V)
-	areCausallyRelatedUsingBFS(hash1: Hash, hash2: Hash): boolean {
+	private _areCausallyRelatedUsingBFS(start: Hash, target: Hash): boolean {
```
Comment (d-roak) [inline packages/object/src/hashgraph/index.ts:226]:
```suggestion
	private _areCausallyRelated(start: Hash, target: Hash): boolean {
```

**Review by d-roak — CHANGES_REQUESTED** [review]
rm useless comments and rename the fns
try to build the code, it might need some refactors because the naming would be different

**Review by JanLewDev — APPROVED** [review]
Clean logic, I like it. I'm slightly confused with the naming as I think the bitset version should be default and BFS version used when memory consumption can become a problem

File: packages/object/src/hashgraph/index.ts (hunk 1 above)
Comment (d-roak) [inline packages/object/src/hashgraph/index.ts:266]:
```suggestion
	areCausallyRelatedUsingBFS(hash1: Hash, hash2: Hash): boolean {
		return (
			this._areCausallyRelatedUsingBFS(hash1, hash2) ||
			this._areCausallyRelatedUsingBFS(hash2, hash1)
		);
	}

```

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**